WordPress/gutenberg
# Move Legacy Widget block to @wordpress/widgets

## Description

Moves the Legacy Widget block from `@wordpress/block-library` → `@wordpress/widgets`.

We've gone back and forth on this a little. It used to be in `@wordpress/edit-widgets`. It ended up in `@wordpress/block-library` for two reasons:

1. **Both the widgets editor (`@wordpress/edit-widgets`) and customizer (`@wordpress/customize-widgets`) need to use the block.**
   
   This is moot now that we have `@wordpress/widgets` which is shared between the two widget editors.
   
   _Solution: Move the block into `@wordpress/widgets.`_

2. **It's potentially useful for users to be able to use this block in the post editor and site editor.**
   
   The problem with this, though, is that most widgets won't work unless you enqueue the scripts and styles that that widget expects to have enqueued. This is a not-insignificant amount of bytes to be loading into the post and site editors for such a niche feature.

   _Solution: Document how a site admin can enable the Legacy Widget block in the post editor if they want it._

## How has this been tested?

I performed these steps while running the plugin against both WordPress trunk and WordPress 5.7:

1. Go to Appearance → Widgets.
2. Insert or select a Legacy Widget block. It should work normally.
3. Go to Appearance → Customize.
4. Insert or select a Legacy Widget block. It should work normally.

## Checklist:
